libyul/optimiser/ControlFlowSimplifier.cpp

```diff
@@ -115,9 +115,51 @@ void ControlFlowSimplifier::operator()(Block& _block)
 	simplify(_block.statements);
 }
 
+void ControlFlowSimplifier::visit(Statement& _st)
+{
+	if (_st.type() == typeid(ForLoop))
+	{
+		ForLoop& forLoop = boost::get<ForLoop>(_st);
+		yulAssert(forLoop.pre.statements.empty(), "");
+
+		size_t outerBreak = m_numBreakStatements;
+		size_t outerContinue = m_numContinueStatements;
+		m_numBreakStatements = 0;
+		m_numContinueStatements = 0;
+
+		ASTModifier::visit(_st);
+
+		if (!forLoop.body.statements.empty())
+		{
+			bool isTerminating = false;
+			TerminationFinder::ControlFlow controlFlow = TerminationFinder::controlFlowKind(forLoop.body.statements.back());
```

Comment:
I guess the case I asked about isn't simplified because this returns `revert` where `numBreaks = 1`.
Shouldn't this use `findFirst...` though?

Reply:
The UNIX philosophy: Each step does only one tiny thing :)